fenix/app/src/androidTest/java/org/mozilla/fenix/ui/DownloadTest.kt

```diff
@@ -238,4 +244,337 @@ class DownloadTest {
+            verifyDownloadPrompt(downloadFile)
+        }.clickDownload {
+        }
+
+        browserScreen {
+        }.openNotificationShade {
+            verifySystemNotificationExists("Download completed")
+            swipeDownloadNotification("Left", true, false)
+            verifySystemNotificationDoesNotExist("Firefox Fenix")
+        }.closeNotificationTray {
+        }
+
+        downloadRobot {
+        }.closeDownloadPrompt {
+            verifyDownloadPromptIsDismissed()
+        }
+    }
+
+    @Test
+    fun stayInPrivateBrowsingPromptTest() {
+        // Clear the "Firefox Fenix default browser notification"
+        notificationShade {
+            cancelAllShownNotifications()
+        }
+
+        downloadFile = "1GB.zip"
+
+        homeScreen {
+        }.togglePrivateBrowsingMode()
+
+        navigationToolbar {
+        }.enterURLAndEnterToBrowser(downloadTestPage.toUri()) {
+            waitForPageToLoad()
+        }.clickDownloadLink(downloadFile) {
+            verifyDownloadPrompt(downloadFile)
+        }.clickDownload {
+        }
+        browserScreen {
+        }.openTabDrawer {
+            closeTab()
+        }
+        browserScreen {
+            verifyCancelPrivateDownloadsPrompt("1")
+            clickStayInPrivateBrowsingPromptButton()
+        }.openNotificationShade {
+            verifySystemNotificationExists("Firefox Fenix")
+        }
+    }
+
+    @Test
+    fun cancelActiveDownloadsFromPrivateBrowsingPromptTest() {
+        // Clear the "Firefox Fenix default browser notification"
+        notificationShade {
+            cancelAllShownNotifications()
+        }
+
+        downloadFile = "1GB.zip"
+
+        homeScreen {
+        }.togglePrivateBrowsingMode()
+
+        navigationToolbar {
+        }.enterURLAndEnterToBrowser(downloadTestPage.toUri()) {
+            waitForPageToLoad()
+        }.clickDownloadLink(downloadFile) {
+            verifyDownloadPrompt(downloadFile)
+        }.clickDownload {
+        }
+        browserScreen {
+        }.openTabDrawer {
+            closeTab()
+        }
+        browserScreen {
+            verifyCancelPrivateDownloadsPrompt("1")
+            clickCancelPrivateDownloadsPromptButton()
+        }.openNotificationShade {
+            verifySystemNotificationDoesNotExist("Firefox Fenix")
+        }
+    }
+
+    // Save edited pdf file from the share overlay
+    @Test
+    fun saveEditedPdfTest() {
+        val genericURL =
+            TestAssetHelper.getGenericAsset(mockWebServer, 3)
+
+        navigationToolbar {
+        }.enterURLAndEnterToBrowser(genericURL.url) {
+            clickPageObject(itemWithText("PDF form file"))
+            fillPdfForm("Firefox")
+        }.openThreeDotMenu {
+        }.clickShareButton {
+        }.clickSaveAsPDF {
+            verifyDownloadPrompt("pdfForm.pdf")
+        }.clickDownload {
+        }.clickOpen("application/pdf") {
+            assertExternalAppOpens("com.google.android.apps.docs")
```

Comment:
See above comments

Reply:
Created a constant, and Oana also created one in her PR.
I'll wait for her PR to get merged and will delete it from my PR.